### PR TITLE
Document handling of special cases when using items

### PR DESCRIPTION
**Pull request by tobiasvl**
Very unsure of the names here. Any better suggestions?

### CODE REVIEW

File: src/constants/memory/wram.asm [hunk 1]
```diff
@@ -784,8 +784,9 @@ wC1AB::
 wC1AC::
   ds 1 ; C1AC
 
-; Whether the A button should do something other than using an item?
-wC1AD::
+; Whether pressing the A button will use the current item, initiate dialog,
```
Comment (tobiasvl) [inline src/constants/memory/wram.asm:787]:
```suggestion
; Whether pressing the A or B button will do something other than using the current item
```

File: src/constants/memory/wram.asm [hunk 2]
```diff
@@ -784,8 +784,9 @@ wC1AB::
...
+; read a sign/interact with a drawer
```
Comment (tobiasvl) [inline src/constants/memory/wram.asm:788]: ; (for example while near an NPC, reading text, etc)

**Comment by daid** [comment]
Is is item usage or button usage?

**Comment by tobiasvl** [comment]
> Is is item usage or button usage?

It's only checked when using items, ie. pressing the A or B button in regular gameplay mode. I guess it's semantics, but it's only checked when using items.

Basically, `$C1AD` is set to a few non-zero values in specific situations, and item handling checks those values for special handling of those situations:

* If near an NPC, when pressing A should initiate dialog, item usage is prohibited and if the sword is up, it's lowered
* If reading non-dialog text (signs, owl statues, etc), item usage is prohibited
* While on the raft, item usage freezes the raft's movement momentarily

File: src/code/bank0.asm [hunk 3]
```diff
@@ -1899,30 +1904,38 @@ CheckItemsToUse::
     ldh  a, [hPressedButtonsMask]                 ; $1242: $F0 $CB
     and  J_B                                      ; $1244: $E6 $20
     jr   z, .nextItemB                            ; $1246: $28 $03
-    ; the two checks from A does not apear here == bug?
```
Comment (tobiasvl) [inline src/code/bank0.asm:1902]: indeed :)

**Comment by daid** [comment]
So it's more a form of an item block, bit like the `hLinkInteractiveMotionBlocked`?

**Comment by tobiasvl** [comment]
> So it's more a form of an item block, bit like the `hLinkInteractiveMotionBlocked`?

Hmm, yes, very similar! More like a hook, maybe, since item usage isn't blocked on the raft, just checked in order to handle it another way. But `wItemUsageBlocked` works fine, or `wLinkItemUsageBlocked` maybe?

I see that the values of `hLinkInteractiveMotionBlocked` don't have names, which they probably should have, so I can't get inspiration from those. On the other hand, a good opportunity to give them similar names, maybe. `LINK_INTERACTIVE_MOTION_BLOCK1` isn't very descriptive though.

**Comment by kemenaran** [comment]
Yup, that could be similar to `hLinkInteractiveMotionBlocked` (whose values are not named yet, because I couldn't figure out how the value `2` is any different from the usual `1`…)

**Comment by tobiasvl** [comment]
It's even _more_ similar to another flag I just found, `wBlockItemUsage` at `$C50A`! That one's used by the shop and trendy game, apparently.

File: src/code/bank0.asm [hunk 4]
```diff
@@ -1899,30 +1904,38 @@ CheckItemsToUse::
...
+
+    ; possibly a minor bug here: we're not
+    ; checking if we're near an NPC or currently
+    ; reading text from a sign, so if the shield
+    ; is equipped to B, we can press and hold B
+    ; at the final text line to close a text box
+    ; and immediately raise the shield. This is
+    ; not possible if the shield is equpped to A.
```
Comment (kemenaran) [inline src/code/bank0.asm:1914]:
```suggestion
    ; not possible if the shield is equipped to A.
```

**Review by kemenaran — COMMENTED** [review]
This PR is quite some work, so how could we resolve the stalling and merge it…

For myself I would be fine with merging it as-this. And otherwise it seems the only pending point is the naming of `wItemUsageMode`, because it is set regardless of whether an item interaction is happening. Maybe `wItemUsageContext`? Or otherwise just as-this :)

**Comment by tobiasvl** [comment]
Yes, sorry, I got very indecisive about the name there :) I like `wItemUsageContext`! I'll use it and rebase.

**Pull request merged**